src/sig/sig.h

```diff
@@ -63,108 +84,123 @@ struct OQS_SIG {
  */
-OQS_API OQS_SIG *OQS_SIG_new(enum OQS_SIG_algid algid);
+OQS_SIG *OQS_SIG_new(const char *method_name);
 
 /**
- * Generates a new signature key pair.
- * @param s                  Pointer to the signature object.
- * @param priv               Pointer where the generated private key will be stored. Caller
- *                           must have allocated s->priv_key_len bytes.
- * @param pub                Pointer where the generated public key will be stored. Caller
- *                           must have allocated s->pub_key_len bytes.
- * @return                   OQS_SUCCESS on success, or OQS_ERROR on failure
+ * Keypair generation algorithm.
+ *
+ * Caller is responsible for allocating sufficient memory for `public_key` and
+ * `secret_key`, based on the `length_*` members in this object or the per-scheme
+ * compile-time macros `OQS_SIG_*_length_*`.
+ *
+ * @param[in] sig The OQS_SIG object representing the signature scheme.
+ * @param[out] public_key The public key represented as a byte string.
+ * @param[out] secret_key The secret key represented as a byte string.
+ * @return OQS_SUCCESS or OQS_ERROR
  */
-OQS_API OQS_STATUS OQS_SIG_keygen(const OQS_SIG *s, uint8_t *priv, uint8_t *pub);
+OQS_STATUS OQS_SIG_keypair(const OQS_SIG *sig, uint8_t *public_key, uint8_t *secret_key);
 
 /**
- * Generates a new signature.
- * @param s         Pointer to the signature object.
- * @param priv      Pointer to the signer's private key, of expected length `s->priv_key_len` bytes.
- * @param msg       Pointer to the message to sign.
- * @param msg_len   Length of the message to sign `msg`.
- * @param sig       Pointer where the generated signature will be stored. Caller must have allocated `s->max_sig_len` bytes.
- * @param sig_len   Pointer to the length of the generated signature.
- * @return          OQS_SUCCESS on success, or OQS_ERROR on failure
+ * Signature generation algorithm.
+ *
+ * Caller is responsible for allocating sufficient memory for `signed_message`,
+ * based on the `length_*` members in this object or the per-scheme
+ * compile-time macros `OQS_SIG_*_length_*`.
+ *
+ * @param[in] sig The OQS_SIG object representing the signature scheme.
+ * @param[out] signature The signature on the message represented as a byte string.
+ * @param[out] signature_len The length of the signature.
+ * @param[in] message The message to sign represented as a byte string.
+ * @param[in] message_len The length of the message to sign.
+ * @param[in] secret_key The secret key represented as a byte string.
+ * @return OQS_SUCCESS or OQS_ERROR
  */
-OQS_API OQS_STATUS OQS_SIG_sign(const OQS_SIG *s, const uint8_t *priv, const uint8_t *msg, const size_t msg_len, uint8_t *sig, size_t *sig_len);
+OQS_STATUS OQS_SIG_sign(const OQS_SIG *sig, uint8_t *signature, size_t *signature_len, const uint8_t *message, size_t message_len, const uint8_t *secret_key);
```

Comment:
`message_len` should be marked as const

Reply:
Yes; leftovers from copy-pasting from nist-branch

Reply:
Well we do it for uint8\_t which can't be changed by the function either, so why one and not the other?

Reply:
we do it for uint8_t _pointers_, the const being a promise to the caller that the function won't modify the content. I don't see why that would be useful for a size parameter.